### DISCUSSION TITLE
Request in worker.events.on('response:mocked', {request} is always null

### DISCUSSION THREAD

**Discussion by canrozanes** [mswjs/msw#1991] [Q&A] [answered]
Hi, 

I was following the docs on [life-cycle events](https://mswjs.io/docs/api/life-cycle-events/) and I was trying to assert on a response after making an API call.

I have the following piece of code that saves all the requests so that I can assert on the responses later.

```
// QUnit  setup
hooks.beforeEach(async function (this: TestContextWithMsw) {
    this.worker = setupWorker(...handlers);
    this.worker.events.on('response:mocked', ({ request, requestId, response }) => {
      this.handledRequests.push({ request, requestId, response });
    });
});
```

I can clearly see that MSW sends a response in the browser network tab but when I add a debugger inside the 'response:mocked' callback, I see that all the request are null. 

![Screenshot 2024-01-22 at 10 39 35 AM](https://github.com/mswjs/msw/assets/37912128/0fe4ceba-aedb-4a47-a3b3-581b414e1e4c)

What could I be missing?

**Comment by kettanaito** [comment, marked as answer]
Hi, @canrozanes. Thanks for letting me know. This does look like a bug. I will convert it into one.
